### PR TITLE
Fixes ray workflow adapter to work with Ray 2.0

### PR DESCRIPTION
**Pull request by elijahbenizzy**
New ray workflows API integration. This breaks when running locally (for me) with a very cryptic RecursionError. Spent some time digging in and I think its something odd with my env?

## Changes
-

## Testing

1.

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Python - local testing

- [ ] python 3.6
- [ ] python 3.7

### CODE REVIEW

**Comment by elijahbenizzy** [comment]
Local error:
```bash
            # TODO(ujvl): Consider how to allow user to retrieve the ready objects.
            values, debugger_breakpoint = worker.get_objects(object_refs, timeout=timeout)
            for i, value in enumerate(values):
                if isinstance(value, RayError):
                    if isinstance(value, ray.exceptions.ObjectLostError):
                        worker.core_worker.dump_object_store_memory_usage()
                    if isinstance(value, RayTaskError):
>                       raise value.as_instanceof_cause()
E                       ray.exceptions.RayTaskError(WorkflowExecutionError): ray::WorkflowManagementActor.execute_workflow() (pid=90345, ip=127.0.0.1, repr=<ray.workflow.workflow_access.WorkflowManagementActor object at 0x11bf64670>)
E                       ray.exceptions.RayTaskError(RecursionError): ray::_workflow_task_executor_remote() (pid=90344, ip=127.0.0.1)
E                         File "/Users/elijahbenizzy/.pyenv/versions/hamilton-fresh/lib/python3.9/site-packages/ray/workflow/task_executor.py", line 115, in _workflow_task_executor_remote
E                           return _workflow_task_executor(
E                         File "/Users/elijahbenizzy/.pyenv/versions/hamilton-fresh/lib/python3.9/site-packages/ray/workflow/task_executor.py", line 84, in _workflow_task_executor
E                           raise e
E                         File "/Users/elijahbenizzy/.pyenv/versions/hamilton-fresh/lib/python3.9/site-packages/ray/workflow/task_executor.py", line 79, in _workflow_task_executor
E                           output = func(*args, **kwargs)
E                         File "/Users/elijahbenizzy/dev/dagworks/hamilton/hamilton/experimental/h_ray.py", line 151, in <lambda>
E                           def check_node_type_equivalence(node_type: typing.Type, input_type: typing.Type) -> bool:
E                         File "/Users/elijahbenizzy/dev/dagworks/hamilton/hamilton/experimental/h_ray.py", line 151, in <lambda>
E                           def check_node_type_equivalence(node_type: typing.Type, input_type: typing.Type) -> bool:
E                         File "/Users/elijahbenizzy/dev/dagworks/hamilton/hamilton/experimental/h_ray.py", line 151, in <lambda>
E                           def check_node_type_equivalence(node_type: typing.Type, input_type: typing.Type) -> bool:
E                         [Previous line repeated 991 more times]
E                       RecursionError: maximum recursion depth exceeded
E
E                       The above exception was the direct cause of the following exception:
E
E                       ray::WorkflowManagementActor.execute_workflow() (pid=90345, ip=127.0.0.1, repr=<ray.workflow.workflow_access.WorkflowManagementActor object at 0x11bf64670>)
E                         File "/Users/elijahbenizzy/.pyenv/versions/3.9.10/lib/python3.9/concurrent/futures/_base.py", line 439, in result
E                           return self.__get_result()
E                         File "/Users/elijahbenizzy/.pyenv/versions/3.9.10/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
E                           raise self._exception
E                         File "/Users/elijahbenizzy/.pyenv/versions/hamilton-fresh/lib/python3.9/site-packages/ray/workflow/workflow_access.py", line 209, in execute_workflow
E                           await executor.run_until_complete(job_id, context, wf_store)
E                         File "/Users/elijahbenizzy/.pyenv/versions/hamilton-fresh/lib/python3.9/site-packages/ray/workflow/workflow_executor.py", line 109, in run_until_complete
E                           await asyncio.gather(
E                         File "/Users/elijahbenizzy/.pyenv/versions/hamilton-fresh/lib/python3.9/site-packages/ray/workflow/workflow_executor.py", line 356, in _handle_ready_task
E                           raise err
E                       ray.workflow.exceptions.WorkflowExecutionError: Workflow[id=test-test_smoke_screen_module] failed during execution.
```

File: hamilton/experimental/h_ray.py [hunk 1]
```diff
@@ -10,6 +11,22 @@
 logger = logging.getLogger(__name__)
 
 
+def raify(fn):
+    """Makes the function into something ray-friendly.
+    This is necessary due to https://github.com/ray-project/ray/issues/28146.
+
+    @param fn: Function to make ray-friendly
+    @return: The ray-friendly version
```
Comment (skrawcz) [inline hamilton/experimental/h_ray.py:19]: wrong formatting. We use `:` . You can instruct pycharm to default to this one.

**Review by skrawcz — CHANGES_REQUESTED** [review]
can you put more info into the commit message please? e.g. ray 2.0 came out and since ray workflows are in alpha the API broke, and now they use `bind` ...

File: hamilton/experimental/h_ray.py (hunk 1 above)
Comment (elijahbenizzy) [reply hamilton/experimental/h_ray.py:19]: Ahh crap knew I missed it

**Review by skrawcz — APPROVED** [review]

**Pull request merged**